File: java/Samples/BiometricStandards/CS/ANTemplateSampleCS/RecordCreateForms/ANType9RecordCreateForm.cs
using System;
using System.IO;
using System.Windows.Forms;
using Neurotec.Biometrics;
using Neurotec.Biometrics.Standards;

namespace Neurotec.Samples.RecordCreateForms
{
	public partial class ANType9RecordCreateForm : ANRecordCreateForm
	{
		#region Public constructor

		public ANType9RecordCreateForm()
		{
			InitializeComponent();

			fromNFRecordPanel.Enabled = rbFromNFRecord.Checked;
			createEmptyPanel.Enabled = rbCreateEmpty.Checked;
		}

		#endregion

		#region Protected methods

		protected override ANRecord OnCreateRecord(ANTemplate template)
		{
			ANType9Record record;
			if (rbFromNFRecord.Checked)
			{
				byte[] nfRecordData;
				try
				{
					nfRecordData = File.ReadAllBytes(tbNFRecordPath.Text);
				}
				catch
				{
					throw new Exception(string.Format("Could not load NFRecord from {0}", tbNFRecordPath.Text));
				}

				NFRecord nfrecord = new NFRecord(nfRecordData);

				record = new ANType9Record(ANTemplate.VersionCurrent, Idc, chbFmtFlag.Checked, nfrecord);
				Template.Records.Add(record);
				return record;
			}
			else
			{
				record = new ANType9Record(ANTemplate.VersionCurrent, Idc)
				{
					ImpressionType = (BdifFPImpressionType)cbImpressionType.SelectedItem,
					MinutiaeFormat = chbFmtFlag.Checked,
					HasMinutiae = chbHasMinutiae.Checked
				};
				record.SetHasMinutiaeRidgeCounts(chbContainsRidgeCounts.Checked, chbHasRidgeCountsIndicator.Checked);
			}

			template.Records.Add(record);
			return record;
		}

		#endregion

		#region Private methods

		private void ANType9RecordCreateFormLoad(object sender, EventArgs e)
		{
			foreach (object value in Enum.GetValues(typeof(BdifFPImpressionType)))
			{
				cbImpressionType.Items.Add(value);
			}
			cbImpressionType.SelectedIndex = 0;
		}

		private void RbFromNFRecordCheckedChanged(object sender, EventArgs e)
		{
			fromNFRecordPanel.Enabled = rbFromNFRecord.Checked;
		}

		private void RbCreateEmptyCheckedChanged(object sender, EventArgs e)
		{
			createEmptyPanel.Enabled = rbCreateEmpty.Checked;
		}

		private void BtnBrowseNFRecordClick(object sender, EventArgs e)
		{
			if (nfRecordOpenFileDialog.ShowDialog() == DialogResult.OK)
			{
				tbNFRecordPath.Text = nfRecordOpenFileDialog.FileName;
			}
		}

		#endregion
	}
}

File: java/Samples/BiometricStandards/VB/ANTemplateSampleVB/RecordCreateForms/ANType9RecordCreateForm.vb
Imports System
Imports System.IO
Imports Neurotec.Biometrics
Imports Neurotec.Biometrics.Standards

Namespace RecordCreateForms

	Partial Public Class ANType9RecordCreateForm
		Inherits ANRecordCreateForm
#Region "Public constructor"

		Public Sub New()
			InitializeComponent()

			fromNFRecordPanel.Enabled = rbFromNFRecord.Checked
			createEmptyPanel.Enabled = rbCreateEmpty.Checked
		End Sub

#End Region

#Region "Protected methods"

		Protected Overrides Function OnCreateRecord(ByVal template As ANTemplate) As ANRecord
			Dim record As ANType9Record
			If rbFromNFRecord.Checked Then
				Dim nfRecordData() As Byte
				Try
					nfRecordData = File.ReadAllBytes(tbNFRecordPath.Text)
				Catch
					Throw New Exception(String.Format("Could not load NFRecord from {0}", tbNFRecordPath.Text))
				End Try

				Dim nfrecord As New NFRecord(nfRecordData)

				record = New ANType9Record(ANTemplate.VersionCurrent, Idc, chbFmtFlag.Checked, nfrecord)
				template.Records.Add(record)
				Return record
			Else
				record = New ANType9Record(ANTemplate.VersionCurrent, Idc) With {.ImpressionType = CType(cbImpressionType.SelectedItem, BdifFPImpressionType), .MinutiaeFormat = chbFmtFlag.Checked, .HasMinutiae = chbHasMinutiae.Checked}
				record.SetHasMinutiaeRidgeCounts(chbContainsRidgeCounts.Checked, chbHasRidgeCountsIndicator.Checked)
			End If

			template.Records.Add(record)
			Return record
		End Function

#End Region

#Region "Private methods"

		Private Sub ANType9RecordCreateFormLoad(ByVal sender As Object, ByVal e As EventArgs) Handles MyBase.Load
			For Each value As Object In System.Enum.GetValues(GetType(BdifFPImpressionType))
				cbImpressionType.Items.Add(value)
			Next value
			cbImpressionType.SelectedIndex = 0
		End Sub

		Private Sub RbFromNFRecordCheckedChanged(ByVal sender As Object, ByVal e As EventArgs) Handles rbFromNFRecord.CheckedChanged
			fromNFRecordPanel.Enabled = rbFromNFRecord.Checked
		End Sub

		Private Sub RbCreateEmptyCheckedChanged(ByVal sender As Object, ByVal e As EventArgs) Handles rbCreateEmpty.CheckedChanged
			createEmptyPanel.Enabled = rbCreateEmpty.Checked
		End Sub

		Private Sub BtnBrowseNFRecordClick(ByVal sender As Object, ByVal e As EventArgs) Handles btnBrowseNFRecord.Click
			If nfRecordOpenFileDialog.ShowDialog() = Windows.Forms.DialogResult.OK Then
				tbNFRecordPath.Text = nfRecordOpenFileDialog.FileName
			End If
		End Sub

#End Region
	End Class
End Namespace
